utils/cq/daos_pylint.py

```diff
@@ -0,0 +1,403 @@
+            if ":avocado:" in msg.msg:
+                continue
+        # Inserting code can cause wrong-module-order.
+        if scons and msg.msg_id == 'C0411' and 'from SCons.Script import' in msg.msg:
+            continue
+
+        if wrapper:
+            vals['path'] = target_file
+            vals['line'] = wrapper.convert_line(msg.line)
+        else:
+            vals['path'] = msg.path
+            vals['line'] = msg.line
+        vals['column'] = msg.column
+        vals['message-id'] = msg.msg_id
+        vals['message'] = msg.msg
+        vals['symbol'] = msg.symbol
+
+        # Duplicates, needed for message_template.
+        vals['msg'] = msg.msg
+        vals['msg_id'] = msg.msg_id
+        vals['category'] = msg.category
+
+        # The build/scons code is mostly clean, so only allow f-string warnings.
+        if scons and msg.symbol != 'consider-using-f-string':
+            vals['category'] = 'error'
+
+        types[vals['category']] += 1
+        symbols[msg.symbol] += 1
+
+        print(args.msg_template.format(**vals))
+
+        if args.format == 'github':
+            if vals['category'] in ('convention', 'refactor'):
+                continue
+            if vals['category'] == 'warning':
+                continue
+            # pylint: disable-next=line-too-long,consider-using-f-string
+            print('::{category} file={path},line={line},col={column},::{symbol}, {msg}'.format(**vals))  # noqa: E501
+
+    if not types or args.reports == 'n':
+        return
+    for (mtype, count) in types.most_common():
+        print(f'{mtype}:{count}')
+
+    for (mtype, count) in symbols.most_common():
+        print(f'{mtype}:{count}')
+
+
+def run_git_files(args):
+    """Run pylint on contents of 'git ls-files'"""
+
+    all_files = FileTypeList()
+
+    ret = subprocess.run(['git', 'ls-files'], check=True, capture_output=True)
+    stdout = ret.stdout.decode('utf-8')
+    for file in stdout.splitlines():
+        all_files.add(file)
+    all_files.run(args)
+
+
+def run_input_file(args, input_file):
+    """Run from a input file"""
+
+    all_files = FileTypeList()
+
+    with open(input_file, encoding='utf-8') as fd:
+        for file in fd.readlines():
+            all_files.add(file.strip())
+
+    all_files.run(args)
+
+
+def main():
+    """Main program"""
+
+    pylinter.MANAGER.clear_cache()
+    parser = argparse.ArgumentParser()
+
+    # Basic options.
+    parser.add_argument('--git', action='store_true')
+    parser.add_argument('--from-file')
+
+    spellings = True
+    try:
+        # pylint: disable-next=import-outside-toplevel,unused-import
+        import enchant  # noqa: F401
+    except ModuleNotFoundError:
+        spellings = False
+    except ImportError:
+        spellings = False
+
+    if spellings:
+        rcfile = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'pylintrc.spellings')
+    else:
+        rcfile = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'pylintrc')
```

Comment:
Is it not easy to accomplish the same thing without having **almost** duplicate files?

Reply:
I wasn't too happy with this either, I can't see a way of having one file and using the command line to enable/disable anything.  What the old code used to do was write modifications to a existing file and I suspect that would be better.

Will try and improve in the next iteration.